vendor/github.com/urfave/negroni/static.go

```diff
@@ -66,6 +66,9 @@ func (s *Static) ServeHTTP(rw http.ResponseWriter, r *http.Request, next http.Ha
 		// redirect if missing trailing slash
 		if !strings.HasSuffix(r.URL.Path, "/") {
 			r.URL.Path += "/"
+			if strings.HasPrefix(r.URL.Path, "//") {
```

Comment:
The url spec seems to indicate that Path should never have a  preceding`//`
https://www.ietf.org/rfc/rfc3986.txt

> If a URI contains an authority component, then the path component
must either be empty or begin with a slash ("/") character. If a URI does not contain an authority component, then the path cannot begin
with two slash characters ("//"). In addition, a URI reference
(Section 4.1) may be a relative-path reference, in which case the
first path segment cannot contain a colon (":") character. The ABNF
requires five separate rules to disambiguate these cases, only one of which will match the path substring within a given URI reference. We use the generic term "path component" to describe the URI substring
matched by the parser to one of these rules.

      path          = path-abempty    ; begins with "/" or is empty
                    / path-absolute   ; begins with "/" but not "//"
                    / path-noscheme   ; begins with a non-colon segment
                    / path-rootless   ; begins with a segment
                    / path-empty      ; zero characters

I wonder if this should apply to the entire path. `dedupeURLSlashes` removes double `/` in the entire path. Which makes sense to me but it could be inadvertently breaking url behavior and potentially goes against the URL spec. 

for example 
/version/x/y/z
and 
/version/x//z 
are technically both valid urls.

Reply:
https://www.ctrl.blog/entry/nginx-requesturi-proxycache.html
This person seems to think it's non-standards-compliant. I'll only merge the first repeat.